lib/ast/function_def.js

```diff
@@ -350,6 +350,47 @@ class FunctionDef extends ExpressionSignature {
             this.is_list, this.is_monitorable, metadata, annotations, this._parent);
     }
 
+    _checkConflictArguments(functions) {
+        const args = {};
+        for (let f of functions) {
+            for (let a of f.args) {
+                if (!(a in args)) {
+                    args[a] = f._argmap[a];
+                    continue;
+                }
+                const err_msg = `Parameter ${a} is defined multiple times in ${functions.map((f) => f.name).join(', ')}`;
+                if (args[a].canonical !== f.getArgCanonical(a))
```

Comment:
Did you mean to call getArgCanonical on both argument? Or do deep equality here?

Reply:
`args[a]` is `ArgumentDef`, and i didn't keep track of function information for simplicity